packages/dds/tree/src/feature-libraries/forest-summary/forestSummarizer.ts

```diff
@@ -164,24 +163,30 @@ export class ForestSummarizer implements Summarizable {
 			0xc21 /* Forest summary content missing in snapshot */,
 		);
 
-		const readAndParseBlob = async <T extends JsonCompatible<IFluidHandle>>(
-			id: string,
-		): Promise<T> => {
-			const treeBuffer = await services.readBlob(id);
-			const treeBufferString = bufferToString(treeBuffer, "utf8");
-			return parse(treeBufferString) as T;
-		};
+		if (await services.contains(forestSummaryMetadataKey)) {
+			const metadata = await readAndParseSnapshotBlob<ForestSummaryMetadata>(
+				forestSummaryMetadataKey,
+				services,
+				parse,
+			);
+			assert(metadata.version === ForestSummaryVersion.v1, "Unsupported forest summary");
+		}
 
 		// Load the incremental summary builder so that it can download any incremental chunks in the
 		// snapshot.
-		await this.incrementalSummaryBuilder.load(services, readAndParseBlob);
+		await this.incrementalSummaryBuilder.load(services, async (id: string) =>
```

Comment:
there are a lot of "id"s in this code. Could you use a more specific name or type? Just looking at this code I don't even know what this async function is (can't see the parameter name for it from the call site).

It seems a bit odd to be adding this async function as a parameter here in this PR: is this part of the versioning logic somehow?